CMakeLists.txt

```
@@ -28,15 +28,41 @@ target_include_directories(argparse INTERFACE
   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>)
 
+
+# Pedantic compiler options
+# Update if necessary
+if(MSVC)
+  # Force to always compile with W4
+  if(CMAKE_CXX_FLAGS MATCHES "/W[0-4]")
+    string(REGEX REPLACE "/W[0-4]" "/W4" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+  else()
+    set(ARGPARSE_PEDANTIC_COMPILE_FLAGS "/W4")
+  endif()
+  list(APPEND ARGPARSE_PEDANTIC_COMPILE_FLAGS "/WX")
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+  set(ARGPARSE_PEDANTIC_COMPILE_FLAGS -Wall -Wno-long-long -pedantic -Wsign-conversion -Wshadow -Wconversion -Werror -Wextra)
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set(ARGPARSE_PEDANTIC_COMPILE_FLAGS -Wall -Wno-long-long -pedantic -Wsign-conversion -Wshadow -Wconversion -Werror -Wextra)
```

I don't see any differences in here, why don't we use this instead:
```suggestion
elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
  set(ARGPARSE_PEDANTIC_COMPILE_FLAGS -Wall -Wno-long-long -pedantic -Wsign-conversion -Wshadow -Wconversion -Werror -Wextra)
```